client/lib/importer/store.js

```diff
@@ -46,168 +43,233 @@ const initialState = Immutable.fromJS( {
 	},
 } );
 
-const equals = ( a, b ) => a === b;
-const increment = a => a + 1;
-
-const removableStates = [ appStates.CANCEL_PENDING, appStates.DEFUNCT ];
-const shouldRemove = importer =>
-	removableStates.some( partial( equals, importer.get( 'importerState' ) ) );
-
-const adjustImporterLock = ( state, { action } ) => {
-	switch ( action.type ) {
-		case IMPORTS_IMPORT_LOCK:
-			return state.setIn( [ 'importerLocks', action.importerId ], true );
-
-		case IMPORTS_IMPORT_UNLOCK:
-			return state.setIn( [ 'importerLocks', action.importerId ], false );
-
-		default:
-			return state;
-	}
-};
+const getImporterItemById = ( state, id ) => get( state, [ 'importers', id ], {} );
 
 const ImporterStore = createReducerStore( function( state, payload ) {
-	let { action } = payload,
-		newState;
+	const { action } = payload;
 
 	switch ( action.type ) {
 		case IMPORTS_STORE_RESET:
 			// this is here to enable
 			// unit-testing the store
-			newState = initialState;
-			break;
+			return initialState;
 
 		case IMPORTS_FETCH:
-			newState = state.setIn( [ 'api', 'isFetching' ], true );
-			break;
+			return {
+				...state,
+				api: {
+					...state.api,
+					isFetching: true,
+				},
+			};
 
 		case IMPORTS_FETCH_FAILED:
-			newState = state
-				.setIn( [ 'api', 'isFetching' ], false )
-				.updateIn( [ 'api', 'retryCount' ], increment );
-			break;
+			return {
+				...state,
+				api: {
+					...state.api,
+					isFetching: false,
+					retryCount: get( state, 'api.retryCount', 0 ) + 1,
+				},
+			};
 
 		case IMPORTS_FETCH_COMPLETED:
-			newState = state
-				.setIn( [ 'api', 'isFetching' ], false )
-				.setIn( [ 'api', 'isHydrated' ], true )
-				.setIn( [ 'api', 'retryCount' ], 0 );
-			break;
+			return {
+				...state,
+				api: {
+					...state.api,
+					isFetching: false,
+					isHydrated: true,
+					retryCount: 0,
+				},
+			};
 
 		case IMPORTS_IMPORT_CANCEL:
 		case IMPORTS_IMPORT_RESET:
-			// Remove the specified importer from the list of current importers
-			newState = state.update( 'importers', importers => {
-				return importers.filterNot(
-					importer => importer.get( 'importerId' ) === action.importerId
-				);
-			} );
-			break;
-
-		case IMPORTS_UPLOAD_FAILED:
-			newState = state
-				.setIn( [ 'importers', action.importerId, 'importerState' ], appStates.UPLOAD_FAILURE )
-				.setIn( [ 'importers', action.importerId, 'errorData' ], {
-					type: 'uploadError',
-					description: action.error,
-				} );
-			break;
-
+			return {
+				...state,
+				importers: {
+					...omit( state.importers, action.importerId ),
+				},
```

Comment:
Would this work as

```javascript
importers: omit( state.importers, action.importerId ),
```

Reply:
Yes! Good catch! I thought I'd gone through and removed those instances but must have missed this one!